framework/Models/HybridModel.py

```diff
@@ -0,0 +1,621 @@
+
+  def localInputAndChecks(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None
+    """
+    Dummy.localInputAndChecks(self, xmlNode)
+    for child in xmlNode:
+      if child.tag == 'settings':
+        self.__readSettings(child)
+      elif child.tag == 'Model':
+        self.modelInstance = child.text.strip()
+        if child.attrib['type'] == 'Code':
+          self.createWorkingDir = True
+      elif child.tag == 'CV':
+        self.cvInstance = child.text.strip()
+      elif child.tag == 'TargetEvaluation':
+        self.targetEvaluationInstance = child.text.strip()
+      elif child.tag == 'ROM':
+        romName = child.text.strip()
+        # 'useRom': [converged, validated]
+        self.romsDictionary[romName] = {'Instance':None,'Converged':False,'Valid':False}
+
+  def __readSettings(self, xmlNode):
+    """
+      Method to read the model settings from XML input files
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None
+    """
+    for child in xmlNode:
+      if child.tag == 'trainStep':
+        self.romTrainStepSize  = int(child.text)
+      if child.tag == 'maxTrainSize':
+        self.romTrainMaxSize = int(child.text)
+      if child.tag == 'initialTrainSize':
+        self.romTrainStartSize = int(child.text)
+      if child.tag == 'tolerance':
+        self.romConvergence = float(child.text)
+      if child.tag == 'selectionMethod':
+        self.modelSelection = child.text.strip()
+
+  def initialize(self,runInfo,inputs,initDict=None):
+    """
+      Method to initialize this model class
+      @ In, runInfo, dict, is the run info from the jobHandler
+      @ In, inputs, list, is a list containing whatever is passed with an input role in the step
+      @ In, initDict, dict, optional, dictionary of all objects available in the step is using this model
+      @ Out, None
+    """
+    if isinstance(self.modelInstance, Models.Model):
+      self.raiseAnError(IOError, "HybridModel has already been initialized, and it can not be initialized again!")
+    self.tempOutputs['uncollectedJobIds'] = []
+    self.modelInstance = self.retrieveObjectFromAssemblerDict('Model', self.modelInstance)
+    self.cvInstance = self.retrieveObjectFromAssemblerDict('CV', self.cvInstance)
+    self.cvInstance.initialize(runInfo, inputs, initDict)
+    self.targetEvaluationInstance = self.retrieveObjectFromAssemblerDict('TargetEvaluation', self.targetEvaluationInstance)
+    if not self.targetEvaluationInstance.isItEmpty():
+      self.raiseAWarning("The provided TargetEvaluation data object is not empty, the existing data will also be used to train the ROMs!")
+      self.existTrainSize = len(self.targetEvaluationInstance)
+    self.tempTargetEvaluation = copy.deepcopy(self.targetEvaluationInstance)
```

Comment:
do you need the tempTargetEvaluation?

Reply:
Yes, I need this to train the ROM. This tempTargetEvaluation only contains the results from the Model runs, and we will not put the results from the ROMs into this data object.